src/progress.jl

```diff
@@ -0,0 +1,110 @@
+import Base: done
+
+export ProgressBar, progress!, msg!, done, right_text!, @progress
+
+type ProgressBar
+  leftText::String
+  rightText::String
```

Comment:
Are there any benefits to storing these properties rather than just sending messages to the frontend? Not sure that it's likely to cause issues, but generally it's a good idea to store any state in one place only.